app.rb

```diff
@@ -0,0 +1,307 @@
+require './item_class'
+require './book'
+require './label'
+require './game_class'
+require './author_class'
+require './music_album'
+require './genre'
+require 'json'
+
+class App # rubocop:disable Metrics/ClassLength
+  attr_accessor :labels, :books, :genres, :music_albums, :authors
+
+  def initialize
+    @labels = []
+    @books = []
+    @genres = []
+    @music_albums = []
+    @authors = []
+    @games = []
+  end
+
+  def params(item)
+    genre = Genre.new(item['genre']['name'])
+    genre.id = item['genre']['id']
+    author = Author.new(item['author']['first_name'], item['author']['last_name'])
+    author.id = item['author']['id']
+    label = Label.new(item['label']['title'], item['label']['color'])
+    label.id = item['label']['id']
+
+    [genre, author, label]
+  end
+
+  def params_exists?
+    abort('There are no genres yet, try adding one! => 10') if @genres.empty?
+    abort('There are no labels yet, try adding one! => 11') if @labels.empty?
+    abort('There are no authors yet, try adding one! => 12') if @authors.empty?
+  end
+
+  def load_albums
+    if File.exist?('music_albums.json')
+      data = JSON.parse(File.read('music_albums.json'))
+      data.each do |album|
+        genre, author, label = params(album)
+
+        music_album = MusicAlbum.new(genre, author, label, album['publish_date'], album['on_spotify'])
+        music_album.id = album['id']
+        @music_albums << music_album
+      end
+    else
+      []
+    end
+  end
+
+  def load_genres
+    if File.exist?('genres.json')
+      data = JSON.parse(File.read('genres.json'))
+      data.each do |genre|
+        my_genre = Genre.new(genre['name'])
+        my_genre.id = genre['id']
+        @genres << my_genre
+      end
+    else
+      []
+    end
+  end
+
+  def load_labels
+    if File.exist?('label.json')
+      data = JSON.parse(File.read('label.json'))
+      data.each do |label|
+        my_label = Label.new(label['title'], label['color'])
+        my_label.id = label['id']
+        @labels << my_label
+      end
+    else
+      []
+    end
+  end
+
+  def load_books
```

Comment:
Please find below the error message received when I try to run your app. Kindly look into that.

![image](https://user-images.githubusercontent.com/10332499/170385043-3e6a4671-0d82-4364-ba42-e94729bd9a25.png)

Reply:
Hello @micheaol, thank you for the review.
We all tested the app on the develop branch in our own machines, one Linux, one wsl, and one windows, it runs perfectly on our end. Could you show us how you got the error, please?

Reply:
@Thi-Ponce ,

Please do you mean I should show you the code on my system?

Reply:
Hello @micheaol , both probably, me, @Kakalanp, and @yersel500  tested everything after the review and didn't get any error even when we erased all the objects and created new ones. Did the error occur when you cloned and used ruby main.rb? We're all very confused about how you got that error. Thank you.

Reply:
@Thi-Ponce ,

The error occurs after I cloned the repo and run `ruby main.rb` command. I don't mean to confuse you guys. I believe the error is a result of the app trying to pass an empty array to the `JSON file`.